staging/shared/restore

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
```

Comment:
Think this script is missing a file extension?

Reply:
I left it without because there was a script in there already without one and because this way we can swap them out for python scripts later without changing the documentation.  Editors _should_ detect the hashbang at the top

Reply:
ok